### ISSUE TITLE
Graph displaying some extra weird lines on Samsung S7 Edge

### ISSUE THREAD

**Issue by sanjay4one** [appsthatmatter/GraphView#499]
Hi,
I am using GraphView library in my android app for real time data plotting. It seems fine on other devices except Samsung S7 edge. An extra line is displaying after my real time data reach the other end. Not able to understand why this is happening. Kindly refer the Screenshot.
![img-20170331-wa0002](https://cloud.githubusercontent.com/assets/6861594/24548361/5314c89e-1632-11e7-9148-b6db6e246f52.jpg)
 Any help would be greatly appreciated.

**Comment by xsoriano** [comment]
Hi, 

I am experiencing a similar behavior on a Samsung J7 running Android 6.01.
There seems to be an additional series data point at the origin of the canvas after using graph.getViewport().setMinX(n); where n is an integer greater than 0. I tried with 0 and negative numbers and the issue is not reproduced. I also tried in a Nexus 5X and the issue is not reproduced.

The extra lines to the origin of the canvas disappears after scrolling or zooming manually.

![screenshot_20180302-135451](https://user-images.githubusercontent.com/11828065/36916380-b79a3202-1e21-11e8-9d56-25f7aae569c3.png)

**Comment by appsthatmatter** [comment]
thx for reporting I still don't have a physical device where I can reproduce it. So help is highly appreciated ;)

**Comment by xsoriano** [comment]
@appsthatmatter Jonas, I went ahead and downloaded the source code to see if I could catch anything, and it seems to me that everything is handled correctly on your side. I see that when the first point of a Line Series is outside of the graph bounds to the left, you interpolate to get the correct position at the updated "zero" coordinate and then you draw a length 0 line for this point calling the renderLine method of the LineGraphSeries.java (line 519). This method calls the Canvas.drawLines(float[] f, Paint p), which unfortunately is undocumented on the Android developer site :/. I went ahead and swapped it for Canvas.drawLine(float startX, float startY, float stopX, float startY, Paint p) and reproduced the same error. This method is documented, and it says that "Degenerate lines (length is 0) will not be drawn", but it seems that it is precisely the place where things are breaking. In other words (this is a wild guess) I think that Canvas.drawLine and Canvas.drawLines are misbehaving for length 0 lines in some cases. I cannot reproduce the problem if I create my own Canvas Activity and try to I isolate Canvas.drawLine and Canvas.drawLines. However, adding a simple check in renderLine (LineGraphSeries.java line 519) such as:
`if (pts.length == 4){
            if (pts[0] == pts[2] && pts[1] == pts[3]){
                return;
            }
}`
seems to take care of the issue (i.e. prevent length 0 lines to be drawn), so I am somewhat confident that the problem lies around there.
I will keep looking when I have some free time. 

Ps. I also verified that the problem can be reproduced in the Samsung Galaxy Tab A (SM-T580) running Android 7.0
